app/Database/Database.hs

```
@@ -35,11 +36,24 @@ setupDatabase quiet = do
     let ind = (T.length dbPath -) . fromMaybe 0 . T.findIndex (=='/') . T.reverse $ dbPath
         db = T.unpack $ T.take ind dbPath
     createDirectoryIfMissing True db
-    runDb (
-        if quiet
-            then void $ runMigrationQuiet migrateAll
-            else runMigration migrateAll
-        )
+
+    -- Ensure SQL schema matches ORM, and initialize schema version table
+    runDb (if quiet
+            then void $ runMigrationQuiet migrateAll >> getDatabaseVersion
+            else runMigration migrateAll >> getDatabaseVersion >> return ())
```

@akarki2005 So you can use the [`void`](https://hackage.haskell.org/package/base-4.21.0.0/docs/Control-Monad.html#v:void) function to throw away the return value of a monadic action (instead of `>> return ()`). But, in my previous comment I missed that the key difference between the branches is the use of `runMigrationQuiet` vs. `runMigration`.

So you can simplify the code further by defining the migration action separately:

```
let migrateAction = if quiet then void . runMigrationQuiet else runMigration
runDb $ void $ ...
```

This is a more functional way about thinking of this problem, where you use a conditional to select a _function_ rather than to actually control program flow.

---

Makes sense, thanks a lot for the feedback David! I'll be sure to point this out at next week's meeting.